modules/message-lane/src/inbound_lane.rs

```diff
@@ -203,19 +236,126 @@ mod tests {
 		});
 	}
 
+	#[test]
+	fn receive_status_update_works_with_batches_from_relayers() {
+		run_test(|| {
+			let mut lane = inbound_lane::<TestRuntime, _>(TEST_LANE_ID);
+			let mut seed_storage_data = lane.storage.data();
+			// Prepare data
+			seed_storage_data.latest_confirmed_nonce = 0;
+			seed_storage_data.latest_received_nonce = 5;
+			seed_storage_data.relayers.push_back((1, 1, TEST_RELAYER_A));
+			// Simulate messages batch (2, 3, 4) from relayer #2
+			seed_storage_data.relayers.push_back((2, 4, TEST_RELAYER_B));
+			seed_storage_data.relayers.push_back((5, 5, TEST_RELAYER_C));
+			lane.storage.set_data(seed_storage_data);
+			// Check
+			assert_eq!(
+				lane.receive_state_update(OutboundLaneData {
+					latest_received_nonce: 3,
+					..Default::default()
+				}),
+				Some(3),
+			);
+			assert_eq!(lane.storage.data().latest_confirmed_nonce, 3);
+			assert_eq!(
+				lane.storage.data().relayers,
+				vec![(4, 4, TEST_RELAYER_B), (5, 5, TEST_RELAYER_C)]
+			);
+		});
+	}
+
 	#[test]
 	fn fails_to_receive_message_with_incorrect_nonce() {
 		run_test(|| {
 			let mut lane = inbound_lane::<TestRuntime, _>(TEST_LANE_ID);
 			assert!(!lane.receive_message::<TestMessageDispatch>(
-				TEST_RELAYER,
+				TEST_RELAYER_A,
 				10,
 				message_data(REGULAR_PAYLOAD).into()
 			));
 			assert_eq!(lane.storage.data().latest_received_nonce, 0);
 		});
 	}
 
+	#[test]
+	fn fails_to_receive_messages_above_max_limit_per_lane() {
+		run_test(|| {
+			let mut lane = inbound_lane::<TestRuntime, _>(TEST_LANE_ID);
+			let max_nonce = <TestRuntime as crate::Trait>::MaxUnconfirmedMessagesAtInboundLane::get();
+			for current_nonce in 1..max_nonce {
```

Comment:
Imo this test fails because it receives 'nonces' in range `[1; max_nonce)` and then with `max_nonce + 1`. So the message with `nonce = max_nonce` => we're trying to receive messages out-of-order => `receive_message` returns false.

So this test isn't actually about `max_limit_per_lane` :) Could you, please, slightly fix it to actually fail because it has more entries than `MaxUnconfirmedMessagesAtInboundLane`? Iiuc the only thing to change is to receive messages from different relayers && then to relay message with `max_nonce` (not `max_nonce + 1`).

Reply:
Thanks for spotting, I corrected the test 👍